spec/dummy/app/models/user.rb

```diff
@@ -1,6 +1,8 @@
 class User < ApplicationRecord
   belongs_to :company
 
+  has_and_belongs_to_many :groups
```

Comment:
never use `has_and_belongs_to_many`, just imagine that method no longer exists
use the has many through approach in its place